scaladoc/src/dotty/tools/scaladoc/tasty/NameNormalizer.scala

```diff
@@ -17,10 +17,9 @@ trait NameNormalizer { self: TastyParser =>
   private val ignoredKeywords: Set[String] = Set("this")
 
   private def escapedName(name: String) =
-    val simpleIdentifierRegex = raw"(?:\w+_[^\[\(\s_]+)|\w+|[^\[\(\s\w_]+".r
+    val simpleIdentifierRegex = "([([{}]) ]|[^A-Za-z0-9$]_)".r
```

Comment:
I think it should be renamed to `complexIdentifierRegex` because now it is the condition that adds backquotes if satisfied. Also, I think we miss there `)` character which also seems to be forbidden